design/datamover.md

```diff
@@ -0,0 +1,67 @@
+- Specifically, CSI snapshots are not guaranteed to be durable or portable, and this is preventing the CSI plugin from going GA.
+- Kubernetes is moving in such a way that every volume would be backed by a CSI driver. CSI snapshotting allows for a way for a user to take a snapshot, but does not guarantee durability nor portability. We want to provide a generic way for users to guarantee portability and durability of these snapshots for a given provider.
+- We also want to allow vendors to provide their own volume data management strategies to be able to work within the velero backup/restore workflow. 
+
+## What is a Data Mover ?
+Data Mover defines an interface that enables movement of kubernetes data from cluster to backup storage(e.g. object storage) and vice-versa.
+
+## Goals
+- The data mover should be **Pluggable**. This means that a user can leverage different types of movers for the same set of resources as long as the movers conform to a similar interface.
+- The interface for Data Mover should support moving data associated with resources **other than** Persistent Volumes
+  - Volume Snapshots are an immediate need, and will likely drive most of the original design discussions.
+  - Ex. a Data Mover for Kubevirt would move VM data to backup storage. The interface should be generic enough to support this use case.
+- The interface for Data Mover should enable publishing its specific configuration parameters for users to have fine-grained control over the movement workflow
+  - Examples:
+    - Data mover for vsphere volume can configure block vs file mount usage
+    - Kubevirt mover can configure export file type
+    - Image mover can specify image format type
+- End-user interaction with data mover should be exposed through the existing backup/restore operations, additionally 2 new CRDs will be introduced - DataMoverBackup
+and DataMoverRestore which would help facilitate data movement and coupling with Velero.
+
+## Non-Goals
+- This implementation will **NOT** change the existing Velero backup/restore workflow.
+  - It is possible in the future that this interface can be used to replace some legacy Velero implementations, but that is **NOT** a goal for this design/implementation.
+- This design of the interface should not include any specific details about (aside from samples):
+  - A mover type
+    - Although we are focused on volumes for phase 1, the interface should have zero knowledge of the underlying data that needs to be moved.
+    - We may provide a way to set the default data mover but still allow users to choose which data mover to user.
+  - Mover implementations
+    - Ex. How a mover handles volumes at the block vs filesystem level is not relevant to the overarching interface design
+    - On the other hand the user will set some mover specific parameters in the interface, thus the user will be aware about some implementation knowledge
+
+## Use Cases
```

Comment:
Shall we move some of the discussion in https://github.com/vmware-tanzu/velero/pull/4768/files#r858431087 
into the use case scenario, like:
1) How the datamovement impacts the backup/restore's statuses?  if a datamovement fails will we mark the backup fail as well?
2)Will the user be monitoring the progress of datamovement via existing `velero backup describe` command?
3)Will we allow user to install/resgister a data mover after velero is installed?

Reply:
@shubham-pampattiwar 
another use case to reflect the discussion in the community meeting, we should allow user to cancel the data movement in some scenarios if the underlying data mover supports that.

Reply:
Done, Update the PR!